diffrax/misc/sde_kl_divergence.py

```diff
@@ -9,8 +9,11 @@
 
 
 def _kl(drift1, drift2, diffusion):
-    inv_diffusion = jnp.linalg.pinv(diffusion)
-    scale = inv_diffusion @ (drift1 - drift2)
+    if diffusion.ndim == 1:
+        scale = (drift1 - drift2) / diffusion
```

Comment:
So my original code here in `sde_kl_divergence` was pretty hacky and not library-ready, and I think it'll still need some more work to get ready.

In particular I think it would make most sense to operate the level of [terms](https://docs.kidger.site/diffrax/api/terms/). This would allow for abstracting over the kind of diffusion used -- e.g. `ControlTerm` versus `WeaklyDiagonalControlTerm` etc. -- rather than the current vector-field-based approach.